ethereum/eip712/eip712.go

```diff
@@ -16,61 +16,100 @@
 package eip712
 
 import (
-	"bytes"
 	"encoding/json"
 	"fmt"
-	"math/big"
-	"reflect"
-	"strings"
-	"time"
 
-	sdkmath "cosmossdk.io/math"
-	"github.com/cosmos/cosmos-sdk/crypto/keys/ed25519"
-	"golang.org/x/text/cases"
-	"golang.org/x/text/language"
-
-	errorsmod "cosmossdk.io/errors"
-	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
-	errortypes "github.com/cosmos/cosmos-sdk/types/errors"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/auth/migrations/legacytx"
 
-	"github.com/ethereum/go-ethereum/common"
-	"github.com/ethereum/go-ethereum/common/math"
+	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/ethereum/go-ethereum/signer/core/apitypes"
+
+	evmtypes "github.com/evmos/ethermint/x/evm/types"
 )
 
+// ConstructUntypedEIP712Data returns the bytes to sign for a transaction.
+func ConstructUntypedEIP712Data(
+	chainID string,
+	accnum, sequence, timeout uint64,
+	fee legacytx.StdFee,
+	msgs []sdk.Msg,
+	memo string,
+) []byte {
+	// TODO: tx.Tip
+	signBytes := legacytx.StdSignBytes(chainID, accnum, sequence, timeout, fee, msgs, memo, nil)
```

Comment:
new tip parameter using nil right now, unless we want to add support here

Reply:
Yes, lets add support - the caller of ConstructUntypedEIP712Data should have the tip data from the tx -- I believe we can just pass it through to this method and use it